### PR TITLE
Add Lua port.

### PR DESCRIPTION
**Pull request by Mustafa32**

### CODE REVIEW

**Comment by christopherdumas** [comment]
Does it work?

File: Ports/lua/oslibc.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
+--- lua/src/loslib.c	2017-04-19 13:29:57.000000000 -0400
++++ /home/mustafa/src/lua/src/loslib.c	2019-05-28 05:44:29.523310628 -0400
+@@ -127,7 +127,7 @@
+ #else				/* }{ */
+ 
+ /* ISO C definitions */
+-#define LUA_TMPNAMBUFSIZE	L_tmpnam
++#define LUA_TMPNAMBUFSIZE	tmpnam
```
Comment (awesomekling) [inline Ports/lua/oslibc.patch]: The POSIX standard says that `L_tmpnam` should be defined by `stdio.h`:
http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/stdio.h.html

I think a better fix would be to edit Serenity's `LibC/stdio.h` to include something like:
```
#define L_tmpnam 256
```

With that fix, you won't need this Lua patch at all :)

File: Ports/lua/Lua.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+function fetch() {
+    run_fetch_web "http://www.lua.org/ftp/lua-5.3.5.tar.gz"
+    run_patch lua.patch -p1
+    run_patch oslibc.patch -p1
+}
+function configure() {
+    run_export_env CC i686-pc-serenity-gcc
+}
+
+function run_make() {
```
Comment (awesomekling) [inline Ports/lua/Lua.sh]: This function is already provided by `.port.include.sh` so you shouldn't replace it here.

**Comment by Mustafa32** [comment]
> 
> 
> Does it work?

It works on my machine.™ Joke aside Lua interpreter works but it has a little bit weird behaviour.I think its Serenity's terminal handling related. Btw i messed up the pull request. (im still git noob.) We only need log2,frexp,frexpf,frexpl functions in math.cpp to work. After recompiling and reinstalling math library Lua port works.

**Comment by awesomekling** [comment]
Did you close this by accident?

**Pull request closed without merging**